### PR TITLE
RDS changes for Keycloak.

### PR DESCRIPTION
**Pull request by MancunianSam**
I've added a variable to set IAM authentication on the datbase with a
default of true. Keycloak is the only thing using this module so that's
OK.

The other changes is to allow the ECS task policy to get the IAM
password for that user.

### CODE REVIEW

File: iam_policy/templates/keycloak_ecs_task_role_policy.json.tpl [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
+        "kms:Decrypt",
+        "kms:GenerateDataKey"
+      ],
+      "Resource": [
+        "${kms_arn}",
+        "arn:aws:rds-db:eu-west-2:${account_id}:dbuser:${cluster_resource_id}/keycloak_user",
+        "arn:aws:sns:eu-west-2:${account_id}:tdr-notifications-${environment}"
+      ]
+    },
+    {
```
Comment (TomJKing) [inline iam_policy/templates/keycloak_ecs_task_role_policy.json.tpl:18]: Is this still needed, as it appears these permissions are in the first statement of the policy?

**Review by TomJKing — COMMENTED** [review]
Looks good. One minor comment.

File: iam_policy/templates/keycloak_ecs_task_role_policy.json.tpl (hunk 1 above)
Comment (MancunianSam) [reply iam_policy/templates/keycloak_ecs_task_role_policy.json.tpl]: Ah good spot, I was trying to merge them and forgot to remove the originals.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**